cmd/analyzer/analyzer.go

```diff
@@ -326,7 +328,7 @@ func NewService(cfg *config.AnalysisConfig) (*Service, error) { //nolint:gocyclo
 			if err1 != nil {
 				return nil, err1
 			}
-			return evmtokens.NewMain(common.RuntimeEmerald, sourceClient, dbClient, logger)
+			return evmtokens.NewAnalyzer(common.RuntimeEmerald, cfg.Analyzers.EmeraldEvmTokens.BatchSize, cfg.Analyzers.EmeraldEvmTokens.StopOnEmptyBatch, cfg.Analyzers.EmeraldEvmTokens.Interval, cfg.Analyzers.EmeraldEvmTokens.InterItemDelay, sourceClient, dbClient, logger)
```

Comment:
I see now why you were asking about grouping constructor params into struct :), this is quite verbose.

I'm in favor of going with ItemBasedAnalyzerConfig here, but your call whether to do it. I think your TODO in the PR description is about that?

Reply:
yeah it's getting hairy - updated to use the cfg struct